networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeMessageHandler.java

```diff
@@ -127,12 +132,31 @@ public void onIncomingMessage(
               if (checkRequestInMinEpochsRange(requestEpoch)
                   && !checkBlobSidecarsAreAvailable(earliestAvailableSlot, endSlot)) {
                 return SafeFuture.failedFuture(
-                    new RpcException.ResourceUnavailableException(
-                        "Requested blob sidecars are not available."));
+                    new ResourceUnavailableException("Requested blob sidecars are not available."));
               }
-              final BlobSidecarsByRangeMessageHandler.RequestState initialState =
-                  new BlobSidecarsByRangeMessageHandler.RequestState(
-                      callback, maxRequestBlobSidecars, startSlot, endSlot);
+
+              final UInt64 finalizedSlot =
+                  combinedChainDataClient.getFinalizedBlockSlot().orElse(UInt64.ZERO);
+
+              final SortedMap<UInt64, Bytes32> canonicalHotRoots;
+              if (endSlot.isGreaterThan(finalizedSlot)) {
+                final UInt64 count = endSlot.minus(finalizedSlot);
```

Comment:
i would call it something like hotSlotsCount to not be confused with request count